libs/pilight/core/json.c

```diff
@@ -1436,3 +1436,28 @@ int json_find_string(JsonNode *object, const char *name, char **out) {
 void json_free(void *a) {
 	free(a);
 }
+
+JsonNode *json_vartype(int vartype) {
```

Comment:
No, the problem here is, that json_vartype can give you an array with more than one type. For example have generic_label `JSON_STRING | JSON_NUMBER` as type for the `label` option. This is why I switched this back from being a switch statement. This would require `json_append_element` to add more than one element at once.